components/json/rs/src/lib.rs

```diff
@@ -212,3 +213,48 @@ pub fn convert_uint64_in_object_array_to_string(path: &str, key: &str, json: &st
 
     serde_json::to_string(&unwrapped_value).unwrap_or_else(|_| "".into())
 }
+
+/// Parses and re-serializes json with all numbers (`u64`/`i64`/`f64`)
+/// converted to strings.
+///
+/// Values other than `u64`/`i64`/`f64` are unchanged. The fields could be
+/// arbitrarily nested, as the conversion is applied to the entire JSON
+/// recursively. Returns an empty String if such conversion is not possible.
+///
+/// Known double fields in the JSON may have `0` as the value, which will be
+/// interpreted as a `u64`. For this reason, `f64` values are also converted to
+/// string to be consistent with how we handle `u64`. Callers must extract the
+/// appropriate values from the string fields using the following Chromium
+/// helpers:
+///   * `u64` -> `base::StringToUint64()`
+///   * `i64` -> `base::StringToInt64()`
+///   * `f64` -> `base::StringToDouble()`
+///
+/// # Arguments
+/// * `json` - A arbitrary JSON string
+///
+/// # Examples
+///
+/// ```js
+/// {"a":1,"b":-1,"c":3.14} -> {"a":"1","b":"-1","c":"3.14"}
+/// {"some":[{"deeply":{"nested":[{"path":123}]}}]} -> {"some":[{"deeply":{"nested":[{"path":"123"}]}}]}
+/// ```
+pub fn convert_all_numbers_to_string(json: &str) -> String {
+    use serde_json::Value;
+
+    fn convert_recursively(json: &mut Value) {
+        match json {
+            Value::Number(n) if n.is_u64() || n.is_i64() || n.is_f64() => {
+                *json = Value::String(n.to_string());
+            }
+            Value::Array(a) => a.iter_mut().for_each(convert_recursively),
```

Comment:
Why are you doing this recursively? This is very inefficient. 
Also why are we applying this to everything instead of just applying it to the specific paths that we know need to be converted? This seems potentially dangerous to me and I think we should be much more strict about how we do conversions here.